### PR TITLE
`HTTPClient`: don't verify 4xx/5xx responses

### PR DESCRIPTION
**Pull request by NachoSoto**

### CODE REVIEW

File: Sources/Misc/Signing+ResponseVerification.swift [hunk 1]
```diff
@@ -65,6 +65,7 @@ extension HTTPResponse where Body == Data {
     ) -> VerificationResult {
         guard let nonce = request.nonce,
               let publicKey = publicKey,
+              !statusCode.isServerError,
```
Comment (tonidero) [inline Sources/Misc/Signing+ResponseVerification.swift:68]: Should we also filter 4xx? Not sure if that would be caught anywhere else

Comment (NachoSoto) [reply Sources/Misc/Signing+ResponseVerification.swift:68]: Yup! Let me add a test for that too 👍🏻

File: Sources/Misc/Signing+ResponseVerification.swift [hunk 2]
```diff
@@ -65,6 +65,7 @@ extension HTTPResponse where Body == Data {
     ) -> VerificationResult {
         guard let nonce = request.nonce,
               let publicKey = publicKey,
+              statusCode.isSuccessfulResponse,
```
Comment (NachoSoto) [inline Sources/Misc/Signing+ResponseVerification.swift:68]: Changed this.

File: Sources/Networking/HTTPClient/HTTPStatusCode.swift [hunk 3]
```diff
@@ -78,7 +78,7 @@ extension HTTPStatusCode: Codable {}
 extension HTTPStatusCode {
 
     var isSuccessfulResponse: Bool {
-        return 200...299 ~= self.rawValue
+        return 200...399 ~= self.rawValue
```
Comment (NachoSoto) [inline Sources/Networking/HTTPClient/HTTPStatusCode.swift:81]: Changed this to match Android.
I confirmed nothing else uses this method, and the only other usage is unaffected.

**Review by NachoSoto — COMMENTED** [review]
Finished this @tonidero

**Review by tonidero — APPROVED** [review]
Looks good!

**Pull request merged**